src/main/java/techreborn/blockentity/machine/tier1/GreenhouseControllerBlockEntity.java

```diff
@@ -0,0 +1,236 @@
+					blockState.scheduledTick((ServerWorld) world, blockPos, world.random);
+				}
+			}
+		}
+		
+		if (block instanceof CropBlock) {
+			// Normal crop
+			CropBlock crop = (CropBlock) block;
+			if (crop.isMature(blockState)) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+					insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos, null));
+					world.setBlockState(blockPos, crop.withAge(0), 2);
+				}
+			}
+		} else if (block instanceof NetherWartBlock) {
+			// Why it isn't a CropBlock? :(
+			if (blockState.get(NetherWartBlock.AGE) >= 3) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+					insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos, null));
+					world.setBlockState(blockPos, blockState.with(NetherWartBlock.AGE, 0), 2);
+				}
+			}
+		} else if (block instanceof SweetBerryBushBlock) {
+			if (blockState.get(SweetBerryBushBlock.AGE) >= 3) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+					insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos, null));
+					world.setBlockState(blockPos, blockState.with(SweetBerryBushBlock.AGE, 1), 2);
+				}
+			}
+		} else if (block instanceof CocoaBlock) {
+			if (blockState.get(CocoaBlock.AGE) >= 2) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+					insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos, null));
+					world.setBlockState(blockPos, blockState.with(CocoaBlock.AGE, 0), 2);
+				}
+			}
+		} else if (block instanceof GourdBlock) {
+			// Pumpkin, Melon
+			if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+				useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+				insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos, null));
+				world.breakBlock(blockPos, false);
+			}
+		} else if (block instanceof SugarCaneBlock
+				|| block instanceof CactusBlock
+				|| block instanceof BambooBlock
+		) {
+			for (int y = 1; world.getBlockState(blockPos.up(y)).getBlock() == block; y++) {
+				if (canUseEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest)) {
+					useEnergy(TechRebornConfig.greenhouseControllerEnergyPerHarvest);
+					insertIntoInv(Block.getDroppedStacks(blockState, (ServerWorld) world, blockPos.up(y), null));
+				}
+				world.breakBlock(blockPos.up(y), false);
+			}
+		}
+		
+	}
+	
+	private void insertIntoInv(List<ItemStack> stacks) {
+		stacks.forEach(stack -> {
+			for (int i = 0; i < 6; i++) {
+				if (insertIntoInv(i, stack).isEmpty()) {
```

Comment:
Might be worth making it only work if it has free space in the inv. as this will just void the items once full I think?

Reply:
Yep, that's correct. That was in one of my TODO lists that i forget about :(